src/relay/backend/contrib/cmsisnn/relay_to_tir.cc

```diff
@@ -32,17 +32,37 @@ namespace relay {
 namespace contrib {
 namespace cmsisnn {
 
-class RelayToTIR : public MixedModeVisitor {
+class RelayToTIRVisitor : public MixedModeVisitor {
  public:
-  explicit RelayToTIR(String func_name) : func_name_(func_name) {}
+  explicit RelayToTIRVisitor(String func_name) : func_name_(func_name) {}
+
+  tir::PrimFunc GetReplacementPrimFunc() { return primfunc_; }
 
  private:
-  void emit_softmax_tir(const Expr& expr) {
+  template <typename T>
+  const T ArgumentToConstantValue(const Expr& arg) {
+    const ConstantNode* constant_node = arg.as<ConstantNode>();
+    return static_cast<const T*>(constant_node->data->data)[0];
+  }
+
+  void CreatePrimFuncForExtern(Array<tir::Var> func_signature,
```

Comment:
A separate PrimFunc maker looks cool. Should we add another flag into it for cases which require intermediate buffer allocations and then under this flag we could emit Allocate?

Reply:
We should check when we get there whether we can keep it generic or if we want to split this into `CreateCallExtern` and `CreatePrimFunc` with a `body`. :thinking: